### PR TITLE
[abandoned]route: Add onion blob to sendtoroute

### LINKED ISSUE #6206: rpc+routing: extend SendToRoute/SendPaymentV2 RPCs with new field allowing custom onion specification
Today whenever we send payments, `lnd` does all the onion construction internally. For certain uses case that implement custom meta-protocols on top of the base BOLT protocol, it can be useful to be able to allow a user to pass in an onion blob they've created themselves. Given the existence of the `lightning-onion` package that houses our onion construction logic, a user can compose that package and this new hypothetical RPC to w/e they want. 

One thing to take note of here is how erorr handling is modified. Today we're able to pinpoint the origin/source of the error as we know the complete path, and can do trial decryption to find out _which_ node encrypted the error. If we're passed in a custom blob instead, then we don't know anything about the path other than the next hop. As a result, users will likely need to implement their _own_ external error handling and mission control updates. Thankfully both are possible based on outstanding PRs, and code that lives in master. 

# Steps To Completion
- [ ] Add a new `OnionBlob` field to the `SendToRoute` RPC call. If present, normal path to onion blob compilation can be skipped 
- [ ] Add a similar field to `SendPaymentV2`. In this case path finding should be _skipped_ all together and the onion blob used in place. Note that since we can't decode the error retries as we handle them today aren't possible. Instead we may need some custom retry hooks, or just decide that this feature will be only added to `SendToRoute`
- [ ] Create provisions in the current APIs to allow for delegated error processing, if we wish to propagate any learnings from the path finding attempt into the internal mission control store. 

Examples of users requesting this in the wild: https://twitter.com/fiatjaf/status/1486468893573099524

### PR DESCRIPTION
**Pull request by ErikEk**
This PR adds the ability to parse a custom onion blob to `sendtoroute`. The blob is limited to 1366 bytes for now.
Partly closes #6206:
- [x] Add a new `OnionBlob` field to the `SendToRoute` RPC call. If present, normal path to onion blob compilation can be skipped.
- [ ] Add a similar field to `SendPaymentV2`. In this case path finding should be skipped all together and the onion blob used in place. Note that since we can't decode the error retries as we handle them today aren't possible. Instead we may need some custom retry hooks, or just decide that this feature will be only added to `SendToRoute`.
- [ ] Create provisions in the current APIs to allow for delegated error processing, if we wish to propagate any learnings from the path finding attempt into the internal mission control store.

### CODE REVIEW

**Review by bitromortac — COMMENTED** [review]
Thank you for working on this, will be a very useful feature! I left a comment in the original issue, happy to know your opinion about it.

**Comment by ErikEk** [comment]
@bitromortac Thank for letting me know. I'll get back to you.

File: cmd/lncli/cmd_payments.go [hunk 1]
```diff
@@ -975,10 +975,22 @@ func sendToRoute(ctx *cli.Context) error {
...
 
+	var onionBlob []byte
+
+	if ctx.IsSet("onion_blob") {
+		onionBlob, err = hex.DecodeString(ctx.String("onion_blob"))
+
+		if err != nil {
+			return err
+		}
+		fmt.Println(onionBlob)
```
Comment (yyforyongyu) [inline cmd/lncli/cmd_payments.go:986]: no need to print?

File: docs/release-notes/release-notes-0.16.0.md [hunk 2]
```diff
@@ -21,6 +21,9 @@ transaction](https://github.com/lightningnetwork/lnd/pull/6730).
   that were signed by our wallet. Prior to this change `SignPsbt` didn't 
   indicate whether the Psbt held any inputs for our wallet to sign.
 
+* Allowing to parse a [custom onion blob though the
```
Comment (yyforyongyu) [inline docs/release-notes/release-notes-0.16.0.md:24]: nit
```suggestion
* Allowing users to parse a [custom onion blob though the
```

**Review by yyforyongyu — COMMENTED** [review]
Will read @bitromortac 's comments and then come back. Just FYI #6683 will largely change the code here.

**Comment by ErikEk** [comment]
This is up for grab, right now, if anyone wants to continue the work.

**Comment by bitromortac** [comment]
!lightninglabs-deploy mute

Disabling the bot until more work is added here.

**Comment by yyforyongyu** [comment]
!lightninglabs-deploy mute 30d

**Comment by SeverinAlexB** [comment]
Hi @ErikEk,
The [LSPSpec LSPS2](https://github.com/BitcoinAndLightningLayerSpecs/lsp/pull/22) might require sending custom onions. Can you give me an update on the state of this PR?

Thanks
Sev

**Pull request closed without merging**